flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,442 @@
+                json={"username": config["username"], "password": config["password"]},
+            )
+
+            if response.status_code == 200 and response.cookies.get("jwt"):
+                self.jwt = response.cookies.get("jwt")
+                logger.debug("Successfully authenticated with Flood.")
+            else:
+                raise PluginError(
+                    f"Failed to authenticate with Flood. (status={response.status_code})"
+                )
+        except RequestException as e:
+            raise PluginError(
+                f"An error occurred while attempting to authenticate with Flood: {e}"
+            )
+
+    def add_torrent_urls(
+        self,
+        config: dict,
+        urls: list,
+        destination: str = None,
+        tags: list = None,
+        start: bool = True,
+    ) -> bool:
+        """
+        Add a list of torrent URLs to Flood.
+        Will only attempt to add torrents if the JWT token is not expired or unset.
+        """
+
+        if not urls:
+            raise PluginError("Parameter 'urls' cannot be empty.")
+        if self.is_jwt_expired():
+            raise PluginError("Not authenticated with Flood.")
+
+        data = {"urls": urls, "start": start}
+
+        if destination:
+            data["destination"] = destination
+        if tags:
+            data["tags"] = tags
+
+        try:
+            response = post(
+                f"{config['url']}/api/torrents/add-urls", json=data, cookies={"jwt": self.jwt}
+            )
+
+            if response.status_code == 200:
+                logger.debug(f"Successfully added {len(urls)} urls to Flood.")
+                return True
+
+            logger.debug(
+                f"Failed to add {len(urls)} urls to Flood. (status={response.status_code})"
+            )
+            return False
+        except RequestException as e:
+            raise PluginError(
+                f"An error occurred while attempting to add torrent urls to Flood: {e}"
+            )
+
+    def add_torrent_files(
+        self,
+        config: dict,
+        files: list,
+        destination: str = None,
+        tags: list = None,
+        start: bool = True,
+    ) -> bool:
+        """
+        Add a list of torrent files to Flood.
+        Each file must be in base64 encoded format.
+        Will only attempt to add torrents if the JWT token is not expired or unset.
+        """
+
+        if not files:
+            raise PluginError("Parameter 'urls' cannot be empty.")
+        if self.is_jwt_expired():
+            raise PluginError("Not authenticated with Flood.")
+
+        data = {"files": files, "start": start}
+
+        if destination:
+            data["destination"] = destination
+        if tags:
+            data["tags"] = tags
+
+        try:
+            response = post(
+                f"{config['url']}/api/torrents/add-files", json=data, cookies={"jwt": self.jwt}
+            )
+
+            if response.status_code == 200:
```

Comment:
I'd consider using `response.ok` instead unless `200` is really the only successful response. Also it kind of muddies the try/except block to have this here. I would move everything but getting the response out.

Reply:
I actually did not know that you can create variables inside a  block and use them outside of the block (without first declaring it outside the block). It makes sense given the way Python works, and that you're not really "creating" variables. 

I was actually going to write here that 200 is the only acceptable answer as it can return 207 which indicates that some succeeded and some failed but it can also respond with 202 which is also acceptable, so I will need to allow only 200 and 202. Good thing I double checked!

The other thing that I did have here was additional logging that a 207 occurred, but did remove it.

Reply:
If that's the case then I'd make a function or method to check the status code so you don't repeat yourself.

Reply:
Introduced new method in Winter/Flexget@79fae7e for creating requests using the task requests session. It requires a list of successful status codes otherwise it will raise a `PluginError` halting the plugin.